Using sklearn's breast cancer dataset to train a Gaussian Naive Bayes model. Related to suggestions by Google's Technical Development Guide. 

Given the origin of this data source, I didn't need to complete any intensive exploratory data analysis.

Developed by Ahmed Kayal 

#### Package imports

In [1]:
import numpy as np
import pandas as pd 
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score

#### Loading data

In [2]:
bc_data = load_breast_cancer()

<br>

Quick review of the dataset's details  

In [3]:
print(bc_data.DESCR[26:1140])



Breast cancer wisconsin (diagnostic) dataset
--------------------------------------------

**Data Set Characteristics:**

    :Number of Instances: 569

    :Number of Attributes: 30 numeric, predictive attributes and the class

    :Attribute Information:
        - radius (mean of distances from center to points on the perimeter)
        - texture (standard deviation of gray-scale values)
        - perimeter
        - area
        - smoothness (local variation in radius lengths)
        - compactness (perimeter^2 / area - 1.0)
        - concavity (severity of concave portions of the contour)
        - concave points (number of concave portions of the contour)
        - symmetry
        - fractal dimension ("coastline approximation" - 1)

        The mean, standard error, and "worst" or largest (mean of the three
        worst/largest values) of these features were computed for each image,
        resulting in 30 features.  For instance, field 0 is Mean Radius, field
        10 is Ra

In [4]:
# Review of data elements
bc_data.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename'])

Assigning variable names based on dictionary keys

In [5]:
# Load feature and specify feature names
features = bc_data.data
feature_names = [i for i in bc_data.feature_names]

# Specify target labels
# For labels, 0 represents malignant tumors and 1 represents benign tumors
target = bc_data.target 
labels = bc_data.target_names

In [6]:
# Create pandas dataframe from input data 
breast_cancer_df = pd.DataFrame(data=features, columns=feature_names)
breast_cancer_df.head()

mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0        17.99         10.38          122.80     1001.0          0.11840   
1        20.57         17.77          132.90     1326.0          0.08474   
2        19.69         21.25          130.00     1203.0          0.10960   
3        11.42         20.38           77.58      386.1          0.14250   
4        20.29         14.34          135.10     1297.0          0.10030   

   mean compactness  mean concavity  mean concave points  mean symmetry  \
0           0.27760          0.3001              0.14710         0.2419   
1           0.07864          0.0869              0.07017         0.1812   
2           0.15990          0.1974              0.12790         0.2069   
3           0.28390          0.2414              0.10520         0.2597   
4           0.13280          0.1980              0.10430         0.1809   

   mean fractal dimension  ...  worst radius  worst texture  worst perimeter  \
0                 0.07871  ...         25.38          17.33           184.60   
1                 0.05667  ...         24.99          23.41           158.80   
2                 0.05999  ...         23.57          25.53           152.50   
3                 0.09744  ...         14.91          26.50            98.87   
4                 0.05883  ...         22.54          16.67           152.20   

   worst area  worst smoothness  worst compactness  worst concavity  \
0      2019.0            0.1622             0.6656           0.7119   
1      1956.0            0.1238             0.1866           0.2416   
2      1709.0            0.1444             0.4245           0.4504   
3       567.7            0.2098             0.8663           0.6869   
4      1575.0            0.1374             0.2050           0.4000   

   worst concave points  worst symmetry  worst fractal dimension  
0                0.2654          0.4601                  0.11890  
1                0.1860          0.2750                  0.08902  
2                0.2430          0.3613                  0.08758  
3                0.2575          0.6638                  0.17300  
4                0.1625          0.2364                  0.07678  

[5 rows x 30 columns]

<br>

#### Data Splitting

I'll use scikit's **train_test_split** function to easily split my original breast cancer dataframe into a training and testing dataframe. 

In [7]:
train, test, train_labels, test_labels = train_test_split(breast_cancer_df[feature_names], target, test_size=.30, random_state=42)


In [8]:
print(train.shape, test.shape)

(398, 30) (171, 30)


<br>

#### Modeling

Now that the data has successfully been split into a train and test set, I'll move onto instantiate the probabilistic classifer. I'll train this model to predict if a tumor is malignant or benign. 

In [9]:
gnb = GaussianNB().fit(train, train_labels)

With my model trained, I'll use it to generate predictions on my test set. I'll then use scikit's **accuracy_score** function to assess this model's accuracy for predicting the type of tumor for each individual. 

In [10]:
test_predictions = gnb.predict(test)
accuracy_score(test_labels, test_predictions)

0.9415204678362573

<br>

**Conclusion** 

From the above accuracy score, the model was able to correctly predict ~94% of the tumor labels in the test set. 